vortex-expr/src/exprs/binary.rs

```diff
@@ -157,12 +190,55 @@ impl AnalysisExpr for BinaryExpr {
                 let min_rhs = self.rhs.min(catalog)?;
                 let max_rhs = self.rhs.max(catalog)?;
 
-                Some(and(eq(min_lhs, max_rhs), eq(max_lhs, min_rhs)))
+                let min_max_check = and(eq(min_lhs, max_rhs), eq(max_lhs, min_rhs));
+
+                // NaN is not captured by the min/max stat, so we must check NaNCount before pruning
+                if let Some(nan_check) = make_nan_predicate(self.lhs(), self.rhs(), catalog) {
+                    Some(and(nan_check, min_max_check))
+                } else {
+                    Some(min_max_check)
+                }
+            }
+            Operator::Gt => {
+                // NaN is not captured by the min/max stat, so we must check NaNCount before pruning
+                let min_max_check = lt_eq(self.lhs.max(catalog)?, self.rhs.min(catalog)?);
+
+                if let Some(nan_check) = make_nan_predicate(self.lhs(), self.rhs(), catalog) {
+                    Some(and(nan_check, min_max_check))
+                } else {
+                    Some(min_max_check)
+                }
+            }
+            Operator::Gte => {
+                // NaN is not captured by the min/max stat, so we must check NaNCount before pruning
+                let min_max_check = lt(self.lhs.max(catalog)?, self.rhs.min(catalog)?);
```

Comment:
This PR also cheekily fixes broken pruning predicate logic, this was `lt_eq` before 🫢